server/utility/minerva_utility.py

```diff
@@ -75,11 +81,23 @@ def findDatasetFolder(currentUser, user, create=False):
 
 def findSharedDatasetFolders(currentUser):
     folderModel = ModelImporter.model('folder')
+    groupModel = ModelImporter.model('group')
+    datasetSharingGroup = groupModel.findOne(query={
+        'name': PluginSettings.DATASET_SHARING_GROUP_NAME
+    })
+    if not datasetSharingGroup:
+        raise AccessException('user group "{0}" doesn\'t exist'.format(
+            PluginSettings.DATASET_SHARING_GROUP_NAME))
+    if datasetSharingGroup['_id'] not in currentUser['groups']:
+        raise AccessException('user doesn\'t belong to user group "{0}"'.format(
+            PluginSettings.DATASET_SHARING_GROUP_NAME))
+
     folders = folderModel.find({
-        'public': True,
+        'public': False,
```

Comment:
Do we need to insist that this *isn't* a public folder.  It seems like we should no longer care if the folder is public or not in this method.

Reply:
You are right. No need to limit to private folders when looking up shared datasets. Code updated.